### PR TITLE
Refactor Card - Email & Phone

### PR DESCRIPTION
**Pull request by HJ-PRK**
Refactor email and phone fields including test files

Fix Checkstyle errors

#55

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 1]
```diff
@@ -168,36 +122,18 @@ public void parse_oneFieldSpecified_success() {
...
+        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder()
+                .withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_FRIEND, VALID_TAG_HUSBAND)
                 .build();
         EditCommand expectedCommand = new EditCommand(targetIndex, descriptor);
 
         assertParseSuccess(parser, userInput, expectedCommand);
     }
 
-    @Test
-    public void parse_invalidValueFollowedByValidValue_success() {
```
Comment (unfazing) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:184]: might we still need this test for question and answer?

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 2]
```diff
@@ -82,55 +70,33 @@ public void parse_invalidPreamble_failure() {
...
-                .withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_AMY).withAddress(VALID_ADDRESS_AMY)
+                .withAddress(VALID_ADDRESS_AMY)
                 .withTags(VALID_TAG_HUSBAND, VALID_TAG_FRIEND).build();
         EditCommand expectedCommand = new EditCommand(targetIndex, descriptor);
 
         assertParseSuccess(parser, userInput, expectedCommand);
     }
 
-    @Test
-    public void parse_someFieldsSpecified_success() {
```
Comment (unfazing) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:123]: might we still need this test for question and answer

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 3]
```diff
@@ -80,22 +71,6 @@ public Card toModelType() throws IllegalValueException {
...
-
-        if (email == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Email.class.getSimpleName()));
-        }
-        if (!Email.isValidEmail(email)) {
-            throw new IllegalValueException(Email.MESSAGE_CONSTRAINTS);
-        }
-        final Email modelEmail = new Email(email);
-
         if (address == null) {
```
Comment (rockman007372) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:74]: @unfazing there is some small factoring issue here - we forgot to changed argument address to answer. I commented so we remember to fix later.

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java (hunk 2 above)
Comment (rockman007372) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:123]: Agree, maybe @HJ-PRK can add it back and comment out?

**Review by rockman007372 — APPROVED** [review]
Looks good to me!

**Pull request merged**